File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.Set;
import java.util.HashSet;
import java.util.Arrays;

class Solution {
    public int solution(int[] elements) {
        
        int n = elements.length;
        int[] extend = new int[n * 2];
        
        for(int i = 0; i < extend.length; i++) {
            if(i < n) {
                extend[i] = elements[i];
            }
            else {
                extend[i] = elements[i - n];
            }
        }
        
        Set<Integer> set = new HashSet<>();
        
        for(int i = 1; i < n+1; i++) {
            for(int j = 0; j < n; j++) {
                int sumNum = Arrays.stream(extend, j, j + i).sum();
                
                set.add(sumNum);
            }
        }
        
        
        return set.size();
    }
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    n = len(elements)
    elements = elements * 2
    
    S = set()
    
    for l in range(1, n+1):
        for start in range(n):
            sum_num = sum(elements[start:start+l])
            
            S.add(sum_num)
    
    return len(S)